### DISCUSSION TITLE
什么时候需要使用简悦 · 同步助手

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#1700] [同步助手]
> 注意：**[简悦 2.2.0](http://ksria.com/simpread/welcome/version_2.2.0.html)** 优化了这个问题，所以当 **扩展端使用同步助手后，同步助手没有开启** 会有如下提示

![image](https://user-images.githubusercontent.com/81074/120885124-e138a500-c619-11eb-9e0f-e344f45f0085.png)

> 虽然可以解决此问题，但仍然建议你看下面的内容。

## 问题描述：

如果你经常遇到以下的提示

![image](https://user-images.githubusercontent.com/77098191/104155110-6f20b300-5421-11eb-9377-6541ce35e1aa.png)

## 同步助手是什么？

[**同步助手**](http://ksria.com/simpread/docs/#/Sync) 是随着简悦 2.1 版本发布的一个 **桌面 App**

**它有两部分构成：**

1. 桌面 App `.app` or `.exe` 文件

![](https://s1.ax1x.com/2020/11/11/BX4SyV.md.png)

2. 扩展端 → 选项页 → 共通 → 同步助手

![image](https://user-images.githubusercontent.com/81074/120885191-2bba2180-c61a-11eb-8067-a3ade846772e.png)

## 为什么要有同步助手？

Chrome 扩展功能受限，所以没有办法做太多事情，如： **保存文件到本地任意位置**，**启动服务** 等。

所以有了同步助手，算做是扩展端在功能上的补充。

## 同步助手能做什么？

它最主要一个功能：[自动同步](http://ksria.com/simpread/docs/#/%E8%87%AA%E5%8A%A8%E5%90%8C%E6%AD%A5) ，使用此功能后，**可以将原本保存到 Chrome 沙箱内部的数据实时保存到本地的任意位置**。

除此以外，还有包括下面的功能：

- [自动同步](http://ksria.com/simpread/docs/#/自动同步)

- [内置解析](http://ksria.com/simpread/docs/#/Sync?id=内置解析)

- [导出服务](http://ksria.com/simpread/docs/#/Sync?id=导出服务)

- [PDF](http://ksria.com/simpread/docs/#/Sync?id=PDF)

- [Epub](http://ksria.com/simpread/docs/#/Sync?id=Epub)

- [pandoc](http://ksria.com/simpread/docs/#/Sync?id=pandoc) 

- [Hypothesis](http://ksria.com/simpread/docs/#/Sync?id=Hypothesis)

- [Readwise.io](http://ksria.com/simpread/docs/#/Sync?id=Readwise)

- [邮件服务](http://ksria.com/simpread/docs/#/Sync?id=邮件服务)

- [发送到Kindle](http://ksria.com/simpread/docs/#/Sync?id=发送到Kindle)

- [系统级快捷键](http://ksria.com/simpread/docs/#/Sync?id=系统级快捷键)

- [打开稍后读](http://ksria.com/simpread/docs/#/Sync?id=打开稍后读)

- [Bookmarklet](http://ksria.com/simpread/docs/#/Sync?id=Bookmarklet)

> 如果你对上述内容并不感兴趣的话，请不要开启此功能，如何关闭请看这里 https://github.com/Kenshin/simpread/discussions/1692

## 自动同步的意义

> 当通过此功能可以把简悦的配置文件（内含稍后读数据）保存到本地的任意位置后，就可以利用网盘实现全网同步功能。

所以，当你没有 **使用网盘同步的需求，则不要开启自动同步功能**，请不要开启此功能，如何关闭请看这里 https://github.com/Kenshin/simpread/discussions/1692

但 **关闭自动同步功能并不影响使用同步助手的其它功能**，如：[Bookmarklet](http://ksria.com/simpread/docs/#/Sync?id=Bookmarklet) [邮件服务](http://ksria.com/simpread/docs/#/Sync?id=邮件服务) 等。

## 自动同步与非自动同步（原有方案）区别

- 前者会把数据保存到本地（借助同步助手）

- 后者会把数据保存到 Chrome 内部

开启 **自动同步** 后，**同步助手负责读取/保存数据**，简悦负责阅读模式或发送稍后读。（所以需要同步助手一直在运行）

**Comment by 973339348** [comment]
![image](https://user-images.githubusercontent.com/43995018/130317680-40370e2a-90f7-4e4e-a516-20161065881c.png)
你好 这个是什么情况呢，英文太多 看着头痛

**Reply by Kenshin** [reply]
这是多个同步助手同时运行的错误提示，请确保只有一个同步助手运行即可。

**Comment by wbx320** [comment]
![image](https://user-images.githubusercontent.com/63857665/130806716-d4b04be1-ec2c-453e-9272-5bb2f03b9617.png)
epub保存不了

**Reply by Kenshin** [reply]
试试安装 https://simpread.ksria.cn/plugins/details/tMGXrU1v0U 这个插件。
另外，使用此功能的话，需要安装 `pandoc`。

最好告知发生问题的 URL。

**Comment by PengBlank** [comment]
明明运行了同步助手，还一直提示这个错误
![image](https://user-images.githubusercontent.com/6948848/134208162-29ff839c-06b8-4293-9238-2304aa3b33e8.png)

**Reply by Kenshin** [reply]
嗨，看看是不是 https://github.com/Kenshin/simpread/discussions/1567 造成的？

**Comment by Simonup** [comment]
<img width="421" alt="1637035658" src="https://user-images.githubusercontent.com/52342861/141904906-8925e0a6-03dd-4be4-a7ae-d58022841161.png">
请问本地配置文件的位置在下载后可以自行移动吗？
另外，配置完坚果云同步后，我在将坚果云文件设为本地同步文件夹时出现了这个对话框，请问如何解决？

**Reply by Kenshin** [reply]
> 配置完坚果云同步后，我在将坚果云文件设为本地同步文件夹时出现了这个对话框，请问如何解决？

应该是你多开同步助手造成的，留意同步助手只能开一个。

> 请问本地配置文件的位置在下载后可以自行移动吗？

没太理解你的意思，简单的说，配置文件在任何位置都无所为，但：

1. 如果你需要使用自动同步功能的话，就需要配置同步助手中配置文件的路径
2. 接第一条，还需要使用第三方同步设备，如坚果云，而你移动的位置也应该在坚果云中。

不过，一般来说 `<坚果云根目录>/SimpRead` 这个文件夹结构不能改动，除非你不使用 API 或授权登录。

**Comment by Ruxinyue** [comment]
一打开同步助手选项页就出不来...
![image](https://github.com/Kenshin/simpread/assets/112150374/06c9dbbf-86ac-4be5-b384-a83c584968e6)

**Reply by Kenshin** [reply]
这个问题一般来说是数据文件损坏，可以使用 http://localhost:7026/database/fixed 修复。

**Comment by ggeniusm3** [comment]
Kenshin你好，我使用Mac打不开同步助手的下载页面，请问如何解决这个问题？

**Reply by Kenshin** [reply]
下载地址 https://experience.sharepoint.cn/:f:/s/simpread/EooPpw5vqrpLoMInbTpV1rgB5hyvrqv-cdhPCGwiLxrDvw?e=NJ9ZPG

另外，你没法访问 https://simpread.pro/download 吗？

**Reply by Kenshin** [reply]
不要在 Email 中回复 😂 请进入 https://github.com/Kenshin/simpread/discussions/1700#discussioncomment-10201186 回复。

我会优化下这个页面，但这几天在休假ing 估计后天会有时间。

**Reply by Kenshin** [reply]
嗨，我优化了这个页面的加载速度，在隐私模式（避免缓存造成的影响）下打开 http://simpread.pro/download 看看能否正常访问？

**Comment by Rocky183** [comment]
明明已经关闭了自动同步功能，为什么每次还会出现这个提示：
<img width="407" alt="1744467088534" src="https://github.com/user-attachments/assets/5e2cb7a6-979a-4652-b4b1-d3b21f0633f9" />

**Reply by Kenshin** [reply]
关闭自动同步后请重启浏览器。
